nepomuk/src/main/java/edu/asu/diging/gilesecosystem/nepomuk/core/files/impl/FileStorageManager.java

```diff
@@ -119,5 +125,35 @@ private boolean createDirectory(String dirPath) {
     public void setFileTypeFolder(String fileTypeFolder) {
         this.fileTypeFolder = fileTypeFolder;
     }
-
+    
+    public String getStoragePath(String username, String uploadId,
+            String documentId) {
+        return baseDirectory + File.separator 
+                + getFileFolderPathInBaseFolder(username, uploadId, documentId);
+    }
+    
+    public void deleteFile(String username, String uploadId, String documentId, String fileName, boolean deleteEmptyFolders) throws NepomukFileStorageException {
+        Path path = Paths.get(getStoragePath(username, uploadId, documentId) + File.separator + fileName);
+        try {
+            Files.delete(path);
+        } catch (NoSuchFileException ex) {
+            throw new NepomukFileStorageException("No such file or directory.", ex);
+        } catch (IOException ex) {
+            throw new NepomukFileStorageException("Could not delete file.", ex);
+        }
+        if (deleteEmptyFolders) {
+            File docFolder = new File(getStoragePath(username, uploadId, documentId));
+            if (docFolder.isDirectory() && docFolder.list().length == 0) {
+                boolean deletedDocFolder = docFolder.delete();
```

Comment:
boolean variable names should not be nouns but words that indicate that it's value is true or false. Here I would just use `deleted`.

Reply:
Also, let's make a comment here explaining that we are using `File.delete()` rather than `Files.delete()` as we don't really care if it was successful or not. If another process put a file in the folder right before we try to delete it, then that's fine. All we want in that case is that the folder does not get deleted. We don't need an exception.